src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -275,6 +275,12 @@ public function initialize(array $query = array(), array $request = array(), arr
         $this->basePath = null;
         $this->method = null;
         $this->format = null;
+
+        // if the request doesn't contain a unique ID, generate a random one
+        if (!$this->headers->has('X_REQUEST_ID')) {
+            $uuid = sprintf('%04x%04x-%04x-%04x-%04x-%04x%04x%04x', mt_rand(0, 0xffff), mt_rand(0, 0xffff), mt_rand(0, 0xffff), mt_rand(0, 0x0fff) | 0x4000,  mt_rand(0, 0x3fff) | 0x8000, mt_rand(0, 0xffff), mt_rand(0, 0xffff), mt_rand(0, 0xffff));
+            $this->headers->set('X_REQUEST_ID', $uuid);
+        }
```

Comment:
@javiereguiluz : please read this post https://benramsey.com/blog/2016/04/ramsey-uuid/ . I don't think UUIDs are as simple as you make it sound like.